docs/versioning.rst

```diff
@@ -0,0 +1,55 @@
+*****************
+Versioning Policy
+*****************
+
+The internal interface of the `ddtrace` library is defined as follows:
+    Any module, function, class or attribute is internal if it is prefixed with a single underscore
+
+    Any package that is named `internal` is internal
+
+    Any package, module, function, class or attribute that is contained within a namespace that is internal is also internal
+
+When a new major version is released it becomes the primary release line, where all new features, bug fixes and security patches land.
+
+The non-negative integers components of the *vMAJOR.MINOR.PATCH* format are incremented by the following rules:
+
+MAJOR
+    API changes incompatible with previous versions
+
+    Functionality changes incompatible with previous versions
+
+    Dropping support for anything, but not limited to, Python versions, library versions supported in integrations, or Datadog APM features.
+
+MINOR
+    API changes that are backwards compatible
+
+    Functionality changes that are backwards compatible
+
+    Adding support for anything, but not limited to, Python versions, library versions supported in integrations, or Datadog APM features.
+
+PATCH
+    Bug fixes that are backwards compatible
+
+    Security fixes that are backwards compatible
+
+
+.. list-table:: Status of Release Lines
+   :widths: 20 30 30 20
+   :header-rows: 1
+
+   * - Release Line
+     - Supported Runtimes
+     - Minimum Trace Agent
```

Comment:
I wonder if this is something we can detect and report on in the library 🤔

Reply:
e.g. on startup try to hit the info endpoint, if we cannot or the response is a version too low, we report that the agent needs to be upgraded?

(necessary for 1.0.0 release?)

Reply:
Let's investigate if this is possible though give users a way to opt out of the guards.